### PR TITLE
feat: add warning diags to aid troubleshooting

### PR DESCRIPTION
**Pull request by johnstcn**
Adds warning diagnostics to aid in troubleshooting issues with cached builds.

### CODE REVIEW

File: internal/provider/cached_image_resource.go [hunk 1]
```diff
@@ -301,12 +305,22 @@ func (r *CachedImageResource) Read(ctx context.Context, req resource.ReadRequest
 	img, err := getRemoteImage(data.Image.ValueString())
 	if err != nil {
 		if !strings.Contains(err.Error(), "MANIFEST_UNKNOWN") {
-			resp.Diagnostics.AddError("Error checking remote image", err.Error())
+			// Explicitly not making this an error diag.
```
Comment (dannykopping) [inline internal/provider/cached_image_resource.go:308]: Why?

File: internal/provider/cached_image_resource.go [hunk 2]
```diff
@@ -365,7 +379,11 @@ func (r *CachedImageResource) Create(ctx context.Context, req resource.CreateReq
 		// FIXME: there are legit errors that can crop up here.
 		// We should add a sentinel error in Kaniko for uncached layers, and check
 		// it here.
-		tflog.Info(ctx, "cached image not found", map[string]any{"err": err.Error()})
+		resp.Diagnostics.AddWarning("Cached image not found!", fmt.Sprintf(
```
Comment (dannykopping) [inline internal/provider/cached_image_resource.go:382]: Nit: do we have to so shouty? Is this a common pattern in tf providers?

**Review by dannykopping — APPROVED** [review]

File: internal/provider/cached_image_resource.go (hunk 2 above)
Comment (johnstcn) [reply internal/provider/cached_image_resource.go:382]: I can remove the bangs!

File: internal/provider/cached_image_resource.go (hunk 1 above)
Comment (johnstcn) [reply internal/provider/cached_image_resource.go:308]: The check for the image not being present is a bit fragile and I'd rather not have a false negative get in the way here.

**Pull request merged**